cmd/version_test.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/exercism/cli/cli"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCurrentVersion(t *testing.T) {
+	expected := fmt.Sprintf("exercism version %s", Version)
+
+	actual := currentVersion()
+	assert.Equal(t, expected, actual)
+}
+
+func TestVersionUpdateCheck(t *testing.T) {
+	fakeEndpoint := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `{"tag_name": "v2.0.0"}`)
+	})
+	ts := httptest.NewServer(fakeEndpoint)
+	defer ts.Close()
+	cli.LatestReleaseURL = ts.URL
+
+	tests := []struct {
+		version  string
+		expected string
+	}{
+		{
+			// It returns the out of date message for versions older than latest.
+			version:  "1.0.0",
+			expected: "Your CLI version is out of date.\nTo upgrade to 2.0.0, see `exercism upgrade`.",
+		},
+		{
+			// It returns the out of date message for pre-release versions.
+			version:  "2.0.0-alpha",
+			expected: "Your CLI version is out of date.\nTo upgrade to 2.0.0, see `exercism upgrade`.",
+		},
```

Comment:
If we want to check that the semver library knows how to deal with prereleases, let's move that test into the tests for UpToDate.